tests/manage/z_cluster/test_log_rotate.py

```diff
@@ -0,0 +1,190 @@
+            3.Write 500M to MGR,MDS,OSD,MON,RGW
+            4.Verify new log created
+            5.Delete logCollector from Storagecluster
+
+        """
+        self.podtype_id = dict()
+        self.podtype_id["mgr"] = [
+            pod.get_mgr_pods,
+            pod.get_ceph_daemon_id(pod.get_mgr_pods()[0]),
+            "ceph-mgr.",
+        ]
+        self.podtype_id["osd"] = [
+            pod.get_osd_pods,
+            pod.get_ceph_daemon_id(pod.get_osd_pods()[0]),
+            "ceph-osd.",
+        ]
+        self.podtype_id["mon"] = [
+            pod.get_mon_pods,
+            pod.get_ceph_daemon_id(pod.get_mon_pods()[0]),
+            "ceph-mon.",
+        ]
+        if config.ENV_DATA["platform"].lower() != constants.AWS_PLATFORM:
+            self.podtype_id["rgw"] = [
+                pod.get_rgw_pods,
+                pod.get_ceph_daemon_id(pod.get_rgw_pods()[0]),
+                "ceph-client.rgw.ocs.storagecluster.cephobjectstore.a",
+            ]
+        self.podtype_id["mds"] = [
+            pod.get_mds_pods,
+            pod.get_ceph_daemon_id(pod.get_mds_pods()[0]),
+            "ceph-mds.",
+        ]
+
+        for pod_type in self.podtype_id:
+            pod_obj = self.get_pod_obj_based_on_id(pod_type)
+            output_cmd = pod_obj.exec_cmd_on_pod(command="ls -l /var/log/ceph")
+            expected_string = (
+                self.podtype_id[pod_type][2]
+                if pod_type == "rgw"
+                else f"{self.podtype_id[pod_type][2]}{self.podtype_id[pod_type][1]}"
+            )
+            self.podtype_id[pod_type].append(
+                len(re.findall(expected_string, output_cmd))
+            )
+        storagecluster_obj = OCP(
+            resource_name=constants.DEFAULT_CLUSTERNAME,
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+            kind=constants.STORAGECLUSTER,
+        )
+        params = '{"spec": {"logCollector": {"enabled": true,"maxLogSize":"500M", "periodicity": "hourly"}}}'
+        storagecluster_obj.patch(
+            params=params,
+            format_type="merge",
+        )
+        log.info("It takes time for storagecluster to update after the edit command")
+        time.sleep(30)
+        log.info("Verify storagecluster on Ready state")
+        verify_storage_cluster()
+
+        for pod_type in self.podtype_id:
+            pod_obj = self.get_pod_obj_based_on_id(pod_type)
+            expected_string = (
+                self.podtype_id[pod_type][2]
+                if pod_type == "rgw"
+                else f"{self.podtype_id[pod_type][2]}{self.podtype_id[pod_type][1]}"
+            )
+            pod_obj.exec_cmd_on_pod(
+                command=f"dd if=/dev/urandom of=/var/log/ceph/{expected_string}.log bs=1M count=530",
+                out_yaml_format=False,
+                container_name="log-collector",
+            )
+
+        for pod_type in self.podtype_id:
+            sample = TimeoutSampler(
+                timeout=963,
```

Comment:
16 mins? does the log rotate action take time to rotate the file after it crosses 500MB?